app/apps/recipes/serializers.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework import serializers
+from apps.recipes.models import Recipe, Ingredient
+
+class IngredientsSerializer(serializers.ModelSerializer):
+
+  class Meta:
+    model = Ingredient
+    fields = ['name']
+
+
+class RecipeSerializer(serializers.ModelSerializer):
+  ingredients = IngredientsSerializer(many=True, allow_empty=False)
+  class Meta:
+    model = Recipe
+    fields = ['id', 'name', 'description', 'ingredients']
+
+  def create(self, validated_data):
+    ingredients_data = validated_data.pop('ingredients')
+    recipe = Recipe.objects.create(**validated_data)
+    for ingredient_data in ingredients_data:
+      Ingredient.objects.create(recipe=recipe, **ingredient_data)
```

Comment:
Suggestion: when creating several objects, Django has a standarized and efficient way of doing it with [`bulk_create`](https://docs.djangoproject.com/en/dev/ref/models/querysets/#bulk-create).

Reply:
```python
def create(self, validated_data):
    ingredients_data = validated_data.pop('ingredients')
    recipe = Recipe.objects.create(**validated_data)

    ingredients = []
    for ingredient_data in ingredients_data:
        ingredient = Ingredient(recipe=recipe, **ingredient_data)
        ingredients.append(ingredient)

    Ingredient.objects.bulk_create(ingredients)
    return recipe
```

Something like that